pkgs/development/libraries/libvirt/default.nix

```diff
@@ -66,45 +66,46 @@ in stdenv.mkDerivation rec {
     substituteInPlace src/lxc/lxc_conf.c \
       --replace 'lxc_path,' '"/run/libvirt/nix-emulators/libvirt_lxc",'
     patchShebangs . # fixes /usr/bin/python references
-    mkdir -p build && cd build
   '';
 
-  configureScript = "../configure";
-
   dontAddDisableDepTrack = true;
 
-  configureFlags = [
-    "--with-runstatedir=/run" # TODO: remove when autoconf 2.70 is released
-    "--localstatedir=/var"
-    "--sysconfdir=/var/lib"
-    "--with-libpcap"
-    "--with-qemu"
-    "--with-vmware"
-    "--with-vbox"
-    "--with-test"
-    "--with-esx"
-    "--with-remote"
-    "--with-polkit"
+  mesonFlags = let
+    opt = option: enable: "-D${option}=${if enable then "enabled" else "disabled"}";
+  in [
+    "-Dauto_features=disabled"
+    "-Drunstatedir=/run"
+    "-Dlocalstatedir=/var"
+    "-Dsysconfdir=/var/lib"
+    "-Dlibpcap=enabled"
+    "-Ddriver_qemu=enabled"
+    "-Ddriver_vmware=enabled"
+    "-Ddriver_vbox=enabled"
+    "-Ddriver_test=enabled"
+    "-Ddriver_esx=enabled"
+    "-Ddriver_remote=enabled"
+    "-Dpolkit=enabled"
+    "-Dbus=enabled"
+    "-Dfuse=disabled" # todo check against previous config
+    "-Dhal=disabled" # todo check against previous config
+    "-Dglusterfs=disabled" # todo check against previous config
+    (opt "storage_iscsi" enableIscsi)
+    (opt "storage_zfs" (stdenv.isLinux && zfs != null))
   ] ++ optionals stdenv.isLinux [
-    "QEMU_BRIDGE_HELPER=/run/wrappers/bin/qemu-bridge-helper"
-    "QEMU_PR_HELPER=/run/libvirt/nix-helpers/qemu-pr-helper"
-    "EBTABLES_PATH=${ebtables}/bin/ebtables-legacy"
-    "CFLAGS=-I${libtirpc.dev}/include/tirpc"
-    "--with-attr"
-    "--with-apparmor"
-    "--with-secdriver-apparmor"
-    "--with-numad"
-    "--with-macvtap"
-    "--with-virtualport"
-    "--with-storage-disk"
-  ] ++ optionals (stdenv.isLinux && zfs != null) [
-    "--with-storage-zfs"
-  ] ++ optionals enableIscsi [
-    "--with-storage-iscsi"
-  ] ++ optionals enableCeph [
-    "--with-storage-rbd"
+    # "QEMU_BRIDGE_HELPER=/run/wrappers/bin/qemu-bridge-helper"
+    # "QEMU_PR_HELPER=/run/libvirt/nix-helpers/qemu-pr-helper"
+    # "EBTABLES_PATH=${ebtables}/bin/ebtables-legacy"
+    # "CFLAGS=-I${libtirpc.dev}/include/tirpc"
+    "-Dattr=enabled"
+    "-Dapparmor=enabled"
+    "-Dsecdriver_apparmor=enabled"
+    "-Dnumad=enabled"
+    "-Dmacvtap=enabled"
+    "-Dvirtualport=enabled"
+    "-Dstorage_disk=enabled"
```

Comment:
We force all features defaulting to auto to enabled:

https://github.com/NixOS/nixpkgs/blob/fd950b9fc7bc97f38b5583bd08e3840834ea6712/pkgs/development/tools/build-managers/meson/setup-hook.sh#L15

so these are mostly redundant:

https://gitlab.com/libvirt/libvirt/-/blob/v6.7.0/meson_options.txt

Reply:
but then I had the feeling that some "auto" were enabled despite missing libraries. Might have been a misunderstanding from my part or a poor meson config (thjat's their first iteration). They have lots of option so maybe it's sensible to disable by default and enable only what we need ?

Reply:
Yeah, I noticed the you are setting the `-Dauto_features=disabled` afterwards. Autotools behave as if `-Dauto_features=auto` was set so force-enabling every feature like we do with Meson must have been quite a shock. Setting `mesonAutoFeatures = "auto"` would be fine for now, though we generally prefer `enabled` since it will fail loudly when a new feature is added so we cannot miss it.

Reply:
I use mesonAutoFeatures